static/src/javascripts/lib/report-error.js

```diff
@@ -0,0 +1,32 @@
+// @flow
+
+/*
+   Report errors to Sentry with optional tags metadata
+   We optionally re-throw the error to halt execution and to ensure the error is
+   still logged to the console via browsers built-in logging for uncaught
+   exceptions. This is optional because sometimes we log errors for tracking
+   user data.
+*/
+
+import raven from 'lib/raven';
+
+type ReportedError = Error & {
+    reported?: boolean,
+};
+
+function reportError(
+    err: ReportedError,
+    tags: Object,
+    shouldThrow: ?boolean = true
```

Comment:
As discussed, `shouldThrow?` would be a better way of indicating that the parameter is optional. `?boolean` indicates that the type of the argument is nullable, which is not strictly true